src/hooks/useCart/useCart.tsx

```diff
@@ -144,7 +148,10 @@ export const useCart = (
           setFeatures(response.features);
         }
         const quotaResponse = await getQuota(ids, authKey, endpoint);
-        if (hasNoQuota(quotaResponse)) {
+        if (hasNegativeQuota(quotaResponse)) {
+          Sentry.captureException("Negative Quota Received");
+          setCartState("NO_QUOTA");
+        } else if (hasNoQuota(quotaResponse)) {
```

Comment:
The negative quota case is really an outlier and practically never happens unless we made a mistake on the backend. Furthermore, the concept of `hasNegativeQuota` is strange in a practical sense, implies that the user owes some items. We should treat the idea of non positive quota as having no quota.

I suggest we move the logic into `hasNoQuota`, so future developers don't need to worry about the concept of "negative quota" because in the first place it should not happen. In a sense, we relegate the notion of "negative quota" to be within the concept of "no quota"

So in the `every` iterator, we can check if it's negative and just log a sentry issue

Reply:
these are good points, thanks!